woocommerce/woocommerce-ios
# Blaze: Update logic checks for banner visibility

Part of: #10098


## Description

This PR fixes the incorrect logic checks for Blaze banner visibility. Previously everything worked because all the checks were the other way around 😓 Unit tests were added to improve confidence.

## Testing instructions

- Log in as an admin to a public WPCom site to ensure permission to create Blaze campaigns. If you have tested this feature, reinstall the app to clear any existing user default values.
- Notice that the Blaze banner is visible on both My Store and Product list screens.
- Dismiss the banner on one of those screens. Notice that the banner is dismissed on both screens.
- The banner should no longer appear on the site even after logging out and in again. The banner should be visible for other eligible sites, though.

## Screenshots

N/A

---
